lookup_plugin/data_gen.py

```diff
@@ -0,0 +1,197 @@
+                dgen_args["seqStart"] = str(json_data['SeqEnd'] + 1)
+                dbicount = str(json_data['TMinDbiFileForForceGC'])
+
+        return dgen_args, dbicount
+
+    def add_params_to_cmd(self, commands, dgen_args):
+        for cmd in commands:
+            if 'vdev' in dgen_args and dgen_args['vdev'] != '':
+                cmd.extend(['-vdev', dgen_args['vdev']])
+            if 'punchwholechunk' in dgen_args and dgen_args['punchwholechunk'] != '':
+                cmd.extend(["-pc", str(dgen_args["punchwholechunk"])])
+            if 'strideWidth' in dgen_args and dgen_args['strideWidth'] != '':
+                cmd.extend(["-sw", dgen_args["strideWidth"]])
+            if 'overlapSeq' in dgen_args and dgen_args['overlapSeq'] != '' and \
+            'numOfSet' in dgen_args and dgen_args['numOfSet'] != '':
+                cmd.extend(["-se", dgen_args["overlapSeq"], "-ts", dgen_args["numOfSet"]])
+        return commands, dgen_args
+
+
+    def run_dummy_generator(self, command):
+        print("command", command)
+        try:
+            result = subprocess.run(command, check=True)
+        except subprocess.CalledProcessError as e:
+            print(f"Error: {e}")
+
+    def generate_data(self, dgen_args, params):
+        bin_path = f'{self.bin_dir}/dummyData'
+        s3_config = f'{self.bin_dir}/s3.config.example'
+        path = f"{self.base_path}/{DBI_DIR}"
+
+        # Create the new directory
+        if not os.path.exists(path):
+            # Create the directory path
+            try:
+                os.makedirs(path, mode=0o777)
+            except Exception as e:
+                print(f"An error occurred while creating '{path}': {e}")
+
+        dbicount = "0"
+        if params['is_random']:
+            dgen_args = self.generate_random_values(dgen_args)
+        else:
+            dgen_args, dbicount = self.set_vals_from_json(dgen_args)
+
+        commands = []
+        for chunk in range(1, params['total_chunks'] + 1):
+            if params['total_chunks'] > 1:
+                dgen_args['chunk'] = str(chunk)
+            command = [bin_path, "-c", dgen_args['chunk'], "-mp", dgen_args['maxPunches'], "-mv", dgen_args['maxVblks'], 
+                       "-p", path, "-pa", dgen_args['punchAmount'], "-pp", dgen_args['punchesPer'], "-ps", dgen_args['maxPunchSize'], 
+                       "-seed", dgen_args['seed'], "-ss", dgen_args['seqStart'], "-t", dgen_args['genType'], "-va", dgen_args['vbAmount'], 
+                       "-l", "5", "-vp", dgen_args['vblkPer'], "-bs", dgen_args['blockSize'], "-bsm", dgen_args['blockSizeMax'], 
+                       "-vs", dgen_args['startVblk'], "-s3config", s3_config, "-s3log", self.s3_upload_log, "-dbic", dbicount]
+            commands.append(command)
+
+        commands, dgen_args = self.add_params_to_cmd(commands, dgen_args)
+
+        if params['is_random']:
+            for cmd in commands:
+                cmd.extend(['-b', 'paroscale-test'])
```

Comment:
define a constant for BUCKET_NAME and use it everywhere

Reply:
also add comments  in data_gen module metioning about the parameters that needs to be passed for each command

Reply:
@RAVAN0407 @NikitaSonsale